src/textual/widgets/_option_list.py

```diff
@@ -0,0 +1,943 @@
+            if option_index == mouse_over:
+                return strip.apply_style(
+                    self.get_component_rich_style(
+                        "option-list--option-hover-highlighted"
+                    )
+                )
+            # Just a normal highlight.
+            return strip.apply_style(
+                self.get_component_rich_style("option-list--option-highlighted")
+            )
+
+        # Perhaps the line is within an otherwise-uninteresting option that
+        # has the mouse hovering over it?
+        if mouse_over is not None and line_number in spans[mouse_over]:
+            return strip.apply_style(
+                self.get_component_rich_style("option-list--option-hover")
+            )
+
+        # It's a normal option line.
+        return strip.apply_style(self.rich_style)
+
+    def scroll_to_highlight(self) -> None:
+        """Ensure that the highlighted option is in view."""
+        highlighted = self.highlighted
+        if highlighted is None:
+            return
+        try:
+            span = self._spans[highlighted]
+        except IndexError:
+            # Index error means we're being asked to scroll to a highlight
+            # before all the tracking information has been worked out.
+            # That's fine; let's just NoP that.
+            return
+        self.scroll_to_region(
+            Region(
+                0, span.first, self.scrollable_content_region.width, span.line_count
+            ),
+            force=True,
+            animate=self.show_vertical_scrollbar,  # https://github.com/Textualize/textual/issues/2077
+        )
+
+    def validate_highlighted(self, highlighted: int | None) -> int | None:
+        """Validate the `highlighted` property value on access."""
+        if not self._options:
+            return None
+        if highlighted is None or highlighted < 0:
+            return 0
+        return min(highlighted, len(self._options) - 1)
+
+    def watch_highlighted(self, highlighted: int | None) -> None:
+        """React to the highlighted option having changed."""
+        if highlighted is not None:
+            self.scroll_to_highlight()
+            if not self._options[highlighted].disabled:
+                self.post_message(self.OptionHighlighted(self, highlighted))
+
+    def action_cursor_up(self) -> None:
+        """Move the highlight up by one option."""
+        if self.highlighted is not None:
+            if self.highlighted > 0:
+                self.highlighted -= 1
+            else:
+                self.highlighted = len(self._options) - 1
+        elif self._options:
+            self.action_first()
+
+    def action_cursor_down(self) -> None:
+        """Move the highlight down by one option."""
+        if self.highlighted is not None:
+            if self.highlighted < len(self._options) - 1:
+                self.highlighted += 1
+            else:
+                self.highlighted = 0
```

Comment:
For your consideration:
```suggestion
            self.highlighted = (self.highlighted + 1) % len(self._options)
```

Reply:
I feel this leaves the code just a little less readable to the casual reader without improving anything.